datafusion/src/execution/context.rs

```diff
@@ -840,6 +859,8 @@ pub struct ExecutionContextState {
     pub config: ExecutionConfig,
     /// Execution properties
     pub execution_props: ExecutionProps,
+    /// Protocol handlers
+    pub protocol_registry: ProtocolRegistry,
```

Comment:
protocol_registry is not being used at the moment right?

Reply:
I made changes after your first review and it's now first used to register all available handlers while the user inits the context in the first place, and get a suitable protocol handler for each path with the scheme.